bin/metrics-alb.rb

```diff
@@ -0,0 +1,171 @@
+         description: 'CloudWatch metric statistics period'
+
+  def cloud_watch
+    @cloud_watch = Aws::CloudWatch::Client.new
+  end
+
+  def loadbalancer
+    @loadbalancer = Aws::ElasticLoadBalancingV2::Client.new
+  end
+
+  def cloud_watch_metric(metric_name, value, load_balancer_name, alb_id)
+    name = ['app', load_balancer_name, alb_id].join('/')
+    cloud_watch.get_metric_statistics(
+      namespace: 'AWS/ApplicationELB',
+      metric_name: metric_name,
+      dimensions: [
+        {
+          name: 'LoadBalancer',
+          value: name
+        }
+      ],
+      statistics: [value],
+      start_time: config[:end_time] - config[:fetch_age] - config[:period],
+      end_time: config[:end_time] - config[:fetch_age],
+      period: config[:period]
+    )
+  end
+
+  def print_statistics(load_balancer_name, alb_id, statistics)
+    result = {}
+    static_value = {}
+    statistics.each do |key, static|
+      r = cloud_watch_metric(key, static, load_balancer_name, alb_id)
+      keys = if config[:scheme] == ''
+               []
+             else
+               [config[:scheme]]
+             end
+      keys.concat [load_balancer_name, key, static]
+      metric_key = keys.join('.')
+
+      static_value[metric_key] = static
+      result[metric_key] = r[:datapoints][0] unless r[:datapoints][0].nil?
+    end
+    result.each do |key, value|
+      output key, value[static_value[key].downcase], value[:timestamp].to_i
+    end
+  end
+
+  def run
+    statistics = {
+      'ActiveConnectionCount'          => 'Sum',
+      'ClientTLSNegotiationErrorCount' => 'Sum',
+      'ConsumedLCUs'                   => 'Sum',
+      'HTTPCode_ELB_4XX_Count'         => 'Sum',
+      'HTTPCode_ELB_5XX_Count'         => 'Sum',
+      'HTTPCode_Target_2XX_Count'      => 'Sum',
+      'HTTPCode_Target_3XX_Count'      => 'Sum',
+      'HTTPCode_Target_4XX_Count'      => 'Sum',
+      'HTTPCode_Target_5XX_Count'      => 'Sum',
+      'IPv6ProcessedBytes'             => 'Sum',
+      'IPv6RequestCount'               => 'Sum',
+      'NewConnectionCount'             => 'Sum',
+      'ProcessedBytes'                 => 'Sum',
+      'RejectedConnectionCount'        => 'Sum',
+      'RequestCount'                   => 'Sum',
+      'RuleEvaluations'                => 'Sum',
+      'TargetConnectionErrorCount'     => 'Sum',
+      'TargetResponseTime'             => 'Average',
+      'TargetTLSNegotiationErrorCount' => 'Sum',
+
+      # The following metrics have a different dimension than the others:
+      # 'HealthyHostCount'               => 'Average',  # Dimension=TargetGroup+LoadBalancer
+      # 'RequestCountPerTarget'          => 'Sum',      # Dimension=TargetGroup
+      # 'UnHealthyHostCount'             => 'Average',  # Dimension=TargetGroup+LoadBalancer
+    }
+
+    begin
+      loadbalancer.describe_load_balancers.load_balancers.each do |alb|
+        next unless config[:albname].nil? or config[:albname] == alb.load_balancer_name
```

Comment:
use `||` over `or` because they are higher priority in ruby http://phrogz.net/programmingruby/language.html#table_18.4